addons/website_sale/static/src/js/website_sale_delivery.js

```diff
@@ -380,8 +384,10 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
      * @param {Event} ev
      */
     _onCarrierClick: async function (ev) {
-        const radio = ev.currentTarget.closest('.o_delivery_carrier_select').querySelector('input[type="radio"]');
-        if (radio.checked) {
+        const radio = ev.currentTarget.closest('.o_delivery_carrier_select').querySelector(
+            'input[type="radio"]'
+        );
+        if (radio.checked && this._arePickupLocationsDisplayed(ev)) {
```

Comment:
As said in my first comment, here the condition applied by `this._arePickupLocationsDisplayed(ev)` isn't good for that case.
I would suggest to keep to redefine the function by something like :

```javascript
_shallDisplayPickupLocations: function (ev) {
    const pickupPointsAreNeeded = ...; // Check o_show_pickup_locations
    const pickupPointsAreDisplayed = ...; // Check  o_list_pickup_locations
    return pickupPointsAreNeeded && !pickupPointsAreDisplayed && !this._isAPickupLocationSelected(ev);
},
```

Reply:
And for sure in that case, the if statement would become
```suggestion
        if (radio.checked && !this._shallDisplayPickupLocations(ev)) {
            return;
        }
```